### DISCUSSION TITLE
Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) v1.6.0

### DISCUSSION THREAD

**Discussion by brkirch** [brkirch/stable-diffusion-webui#38] [Announcements]
This is an offline standalone installer for a slightly modified version of AUTOMATIC1111's Stable Diffusion web UI (changes were applied to [AUTOMATIC1111's Stable Diffusion Web UI v1.6.0](https://github.com/AUTOMATIC1111/stable-diffusion-webui/releases/tag/v1.6.0)). The [ControlNet extension](https://github.com/Mikubill/sd-webui-controlnet) (and its dependencies) as of Mikubill/sd-webui-controlnet@fd37e9fc7ced2c3a39aaa3860916672c8d0fbfe8 is included. All dependencies needed to start using Stable Diffusion (other than the model) are included and this does _not_ require Xcode command line tools to be installed.

macOS 13.5 or later is highly recommended, older versions of macOS are untested.

<hr /><em>This discussion was created from the release <a href='https://github.com/brkirch/stable-diffusion-webui/releases/tag/v1.6.0-mac-builds'>Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) v1.6.0</a>.</em>

**Comment by Max1mu1** [comment]
Hi! How to run dreambooth on mac m1 is there such an opportunity? expected in the future?

**Reply by brkirch** [reply]
I haven't tried using DreamBooth recently, but I might look into it after I finish doing some other (somewhat unrelated) performance optimizations.

**Comment by iliavls** [comment]
May I use electron based gui with this version? On mac M1 it was very helpful. Also thank you for your work!

**Reply by brkirch** [reply]
The `—electron` option will only be in the experimental branch for the moment. I’ll probably add the option in future stable releases.

**Reply by jrittvo** [reply]
Are you planning on releasing 2 tracks now?  One based on `master`, and the other based on `dev` as the experimental track?

**Reply by brkirch** [reply]
Branch `mac-builds-experimental` is synced with `dev` (those are the releases with dates in the release title and tag name), branch `mac-builds` is synced with `master`.

Going forward I'm going to try to do a release from `mac-builds` for every version update. I may sometimes also do an additional release shortly after that to update the ControlNet extension.

On the other hand, releases from `mac-builds-experimental` have no planned schedule. I will put out a new release whenever I feel there have been enough changes in `dev` to justify it, or if I want to add an additional feature, bug fix, or otherwise change something.

**Comment by cre8ivediffusion** [comment]
Not working.

<img width="590" alt="Screenshot 2023-09-06 at 1 39 01 PM" src="https://github.com/brkirch/stable-diffusion-webui/assets/122299976/4e6e64e7-e743-455a-bc7e-8f5657e8457c">

**Reply by brkirch** [reply]
Click on the installer while holding control on the keyboard and select “Open” from the menu that appears.

**Reply by OrganicBeej** [reply]
Hi,

Updated to iOS 13.5.2 just now and followed above instructions but this time, I get this message?
<img width="552" alt="image" src="https://github.com/brkirch/stable-diffusion-webui/assets/77268953/cb3f4989-7337-4946-bc23-ca278c25fc8f">

**Reply by OrganicBeej** [reply]
Is this asking for new/update install for A1111? I dragged the d-loaded folder (from your zipped link) and renamed it stable-diffusion-webui where the normal one is (and re-named my recent one).

So I guess it's 'Update Existing Install'?

Thanks :)

**Reply by OrganicBeej** [reply]
OK, no worries, I have successfully got it installed & running and wow! Swap is literally 256k and is usually much more and Python is only 4.85GB. 🤓😮⭐️⭐️⭐️ 

I will load some ControlNet Models for 1.5 & SDXL plus, try the swapping models to see if it crashes Python. But so far, so good! :)

**Comment by misato2099** [comment]
Can it support SDXL models now?

**Reply by jrittvo** [reply]
Yes. SDXL base and refiner models, with controls (options/settings) for each.

**Reply by misato2099** [reply]
I cannot import SDXL models, I tried sd_xl_base_1_0_0_9vae_clip_vit_l14_f16.ckpt, sd_xl_base_1_0_0_9vae_f16.ckpt, and sd_xl_base_1_0_0_9vae_open_clip_vit_bigg14_f16.ckpt. I installed it by updating existing installation. Do I need to uninstall it first? Thank you!

**Reply by Max1mu1** [reply]
Why do you use ckpt? Try safetensors it works fine.

**Reply by misato2099** [reply]
Now it's fine when I tried another model in safetensors format. Thank you so much! BTW, is the old controlnet (i.e.: openpose, linearts, depth) not available to SDXL?

**Reply by Max1mu1** [reply]
Yeap, right it is not available. There is a models for SDXL but it has not all standart preprocessors

**Comment by senshock** [comment]
<img width="683" alt="image" src="https://github.com/brkirch/stable-diffusion-webui/assets/139965601/7068979e-947c-43f5-9fdb-6b22a0942f4c">
When I installed webUI and opened it, webUI could run normally, but when I started it for the second time, this error occurred. Next, I installed it using pip3 install packaging, but the problem still exists.

**Comment by misato2099** [comment]
It seems it cannot run after I update my Mac to MacOS 14, which the error shown as following when I generated image:

-[IOGPUMetalCommandBuffer validate]:215: failed assertion `commit an already committed command buffer'
./webui.sh: line 270:  3445 Abort trap: 6           "${python_cmd}" -u "${LAUNCH_SCRIPT}" "$@"
